src/pages/index.js

```diff
@@ -2,76 +2,226 @@ import "./index.css";
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo(data);
+    // a = true
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      deleteConfirm.renderDelete(true);
+      deleteConfirm.close();
+      card.remove();
+    })
+    .catch(() => {
+      deleteConfirm.renderDelete(false);
+    })
+    .finally((err) => {
+      deleteConfirm.close();
+      console.log(err);
+    });
+
+const handleLikeClick = (card) => {
+  console.log("test");
+  api
+    .toggleCardLike(card._id, card._isLiked)
+    .then((cardData) => {
+      card.getLikeStatus(cardData.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+};
+
+function handleChangePicture(avatar) {
+  changePictureModal.setEventListeners();
+  changePictureModal.renderLoading(true);
+  api
+    .setUserAvatar(avatar)
+    .then((data) => {
+      console.log(data.avatar);
+      userInfo.setUserAvatar(data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      changePictureModal.renderLoading(false);
+      changePictureModal.close();
+    });
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  handleChangePicture
+);
+changePictureModal.setEventListeners();
+const pictureFormValidator = new FormValidator(config, pictureChangeForm);
+pictureFormValidator.enableValidation();
+profilePicture.addEventListener("click", () => {
+  console.log("not working");
+  pictureFormValidator.resetValidation();
+  changePictureModal.open();
+});
+
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true);
+    console.log(userData);
+    api
+      .saveUserData(
+        { name: userData.title, about: userData.description },
+        userData.avatar
+      )
+      .then((res) => {
+        handleSetUserInfo(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        editProfileModal.renderLoading(false);
+      });
+    function handleSetUserInfo(userData) {
```

Comment:
**Could be improved**

`handleSetUserInfo`


it's just `setUserInfo` (without `handle`)